### ISSUE TITLE
Changes in annotation of services are not propagated instantly

### ISSUE THREAD

**Issue by gecube** [gravitational/teleport#36181]
In addition to #36179

Expected behavior:

All changes to service names and annotations are propagated to Teleport WebUI instantly. It could be leveraged by using WATCH approach in teleport agent application discovery service

Current behavior:

I need to restart teleport agent to get updated view in Teleport WebUI with a command like

```
kubectl delete -n teleport-system pods --all
```

**Comment by tigrato** [comment]
Teleport performs service scraping every 5 minutes, and the updates will be visible in the subsequent refresh cycle, eliminating the need to delete pods.

Although we considered watchers, we opted against them due to the API server's frequent termination of watchers with random timeouts to distribute load evenly across various etcd servers. Implementing re-connection logic would be necessary in this case.

Do you find waiting for 5 minutes too lengthy?

**Comment by gecube** [comment]
@tigrato Hi! Yes, I want to get the changes in 1 minute max.

**Comment by tigrato** [comment]
> @tigrato Hi! Yes, I want to get the changes in 1 minute max.

Can you describe your usecase for this? 
What's the reasoning behind 1m refresh time?

**Comment by gecube** [comment]
@tigrato Let's say - I am releasing a new version of application. With https mandatory. Obviously, teleport will think that it still has the all parameters for the service. And wouldn't connect (I faced the same issue with kibana service - it uses by default TLS with self-signed certificate, so it's impossible to connect it via teleport).
The second thing - that there is no light or status in the teleport WebUI which could give me idea about service synchronisation state. So it is possible to run into the situation that I redeployed some application, service name was changed, the proxying is not working, and the end user must wait as much as 5 minutes to get new service available
